vars/dockerExecute.groovy

```diff
@@ -1,51 +1,78 @@
 import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.ConfigurationMerger
 
 def call(Map parameters = [:], body) {
 
     def STEP_NAME = 'dockerExecute'
     def PLUGIN_ID_DOCKER_WORKFLOW = 'docker-workflow'
 
-    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters){
-        def dockerImage = parameters.dockerImage ?: ''
-        Map dockerEnvVars = parameters.dockerEnvVars ?: [:]
-        def dockerOptions = parameters.dockerOptions ?: ''
-        Map dockerVolumeBind = parameters.dockerVolumeBind ?: [:]
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+        final script = parameters.script
+        Map stepConfig = ConfigurationLoader.stepConfiguration(script, 'kubernetes')
 
-        if(dockerImage) {
+        Set parameterKeys = ['dockerImage',
+                             'dockerOptions',
+                             'dockerWorkspace',
+                             'dockerEnvVars',
+                             'dockerVolumeBind']
 
-            if (! isPluginActive(PLUGIN_ID_DOCKER_WORKFLOW) ) {
-                echo "[WARNING][${STEP_NAME}] Docker not supported. Plugin '${PLUGIN_ID_DOCKER_WORKFLOW}' is not installed or not active. Configured docker image '${dockerImage}' will not be used."
-                dockerImage = null
+        Set stepConfigKeys = ['jnlpAgent',
+                              'imageToContainerMap']
+
+        Map config = ConfigurationMerger.merge(parameters, parameterKeys, stepConfig, stepConfigKeys)
```

Comment:
Use the config loader as in other steps: https://github.com/SAP/jenkins-library/blob/master/vars/snykExecute.groovy#L27